src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java

```diff
@@ -323,16 +323,31 @@ private void getChanges2(ChangeEventSourceContext context,
                         final String tabletId = entry.getValue();
                         YBPartition part = new YBPartition(tabletId);
                       if (snapshotCompleted.size() == tabletPairList.size()) {
-                        LOGGER.info("Snapshot completed for all the tablets! Stopping.");
-                        break;
+                        LOGGER.debug("Snapshot completed for all the tablets! Stopping.");
+                        if (!snapshotter.shouldStream()) {
+                            // This block will be executed in case of initial_only mode
+                            LOGGER.info("Snapshot completed for initial_only mode, stopping the connector now");
+                            return;
+                        }
                       }
+
+
+                      OpId cp = snapshotter.shouldSnapshot() ? offsetContext.snapshotLSN(tabletId) : offsetContext.lsn(tabletId);
```

Comment:
Is the ternary operator required here as `cp` is set again in line 340 if snapshot is completed. Will it suffice to only set it to `snapshotLSN` in this line?

Reply:
Actually, yes. If we remove the ternary operator then in case of `snapshot.mode=never`, we will never reach to line 340 and we will end up taking the snapshot of the table.

Note that I would simply call this a "hack" to accommodate the snapshot and streaming in a single class, future plan is to have the logic in separate classes to be in sync with Debezium's way of things.